File: API/Entities/Post.cs
using System.ComponentModel.DataAnnotations.Schema;
namespace API.Entities
{
    [Table("Posts")]
    public class Post
    {

        public int Id { get; set; }
        public string Title { get; set; }
        public string Status { get; set; }
        public string Content { get; set; }
        public string Type { get; set; } // Programming, Travel ETC.
        public int Upvotes { get; set; }
        public DateTime Created { get; set; } = DateTime.UtcNow;
        public DateTime LastUpdated { get; set; } = DateTime.UtcNow;
        public int UserId { get; set; }
        public User User { get; set; }
    }
}

File: src/client/src/app/_models/post.ts
export interface Post {
    id: number;
    title: string;
    desc: string;
    status: string;
    content: string;
    type: string;
    upvotes: number;
    linkForPost: string;
    created: Date;
    lastUpdated: Date;
    userId: number;
    userName: string;
}
